tests/e2e-test-framework/framework/utils.py

```diff
@@ -508,7 +574,113 @@ def annotate_pvc(
+            ]:
+                self.custom_objects_api.delete_collection_cluster_custom_object(
+                    group=const.CR_GROUP,
+                    version=const.CR_VERSION,
+                    plural=plural,
+                    grace_period_seconds=0,
+                    propagation_policy="Foreground",
+                )
+                logging.info(f"CR {plural}: delete request sent")
+            self.core_v1_api.delete_collection_namespaced_persistent_volume_claim(
+                namespace=namespace
+            )
+            logging.info("waiting for resources to be in the removing state")
+            time.sleep(10)
+            lvg_list = self.custom_objects_api.list_cluster_custom_object(
+                group=const.CR_GROUP,
+                version=const.CR_VERSION,
+                plural="logicalvolumegroups",
+            )["items"]
+            for lvg in lvg_list:
+                if "finalizers" in lvg.get("metadata", {}):
+                    lvg["metadata"]["finalizers"] = []
+                    self.custom_objects_api.replace_cluster_custom_object(
+                        group=const.CR_GROUP,
+                        version=const.CR_VERSION,
+                        namespace=namespace,
+                        plural=const.LVG_PLURAL,
+                        name=lvg["metadata"]["name"],
+                        body=lvg,
+                    )
+            for v in self.list_volumes():
+                if "finalizers" in v.get("metadata", {}):
+                    v["metadata"]["finalizers"] = []
+                    self.custom_objects_api.replace_namespaced_custom_object(
+                        const.CR_GROUP,
+                        const.CR_VERSION,
+                        namespace,
+                        plural=const.VOLUMES_PLURAL,
+                        name=v["metadata"]["name"],
+                        body=v,
+                    )
+        except ApiException as e:
+            print(
+                f"Exception when calling CustomObjectsApi->delete_namespaced_custom_object: {e}"
+            )
+
+    def recreate_pod(self, pod: V1Pod) -> V1Pod:
+        """
+        Recreates a Kubernetes Pod by deleting the existing Pod and waiting for a new Pod to be created.
+
+        Args:
+            pod (V1Pod): The Pod to recreate.
+
+        Returns:
+            V1Pod: The recreated Pod.
+        """
+        namespace = pod.metadata.namespace
+        labels = pod.metadata.labels if pod.metadata.labels is not None else {}
+        label_value = labels.get("app", None)
+
+        assert label_value is not None, "pod label 'app' not found"
+        self.core_v1_api.delete_namespaced_pod(
+            name=pod.metadata.name, namespace=namespace
+        )
+        logging.info(
+            f"pod {pod.metadata.name} deleted, waiting for a new pod to be created"
+        )
+
+        time.sleep(5)
+        pod = self.list_pods(label="app=" + label_value, namespace=namespace)[
```

Comment:
the pod name will be the same before and after delete process, right? do we need this code?

Reply:
the name will stay the same, however, a resource UID is going to change, it might be useful for the event inspection

Reply:
done